shared/contracts/BaseTemplate.sol

```diff
@@ -32,6 +32,11 @@ contract BaseTemplate is APMNamehash, IsContract {
     bytes32 constant internal FINANCE_APP_ID = 0xbf8491150dafc5dcaee5b861414dca922de09ccffa344964ae167212e8c673ae;
     bytes32 constant internal TOKEN_MANAGER_APP_ID = 0x6b20a3010614eeebf2138ccec99f028a61c811b3b1a3343b6ff635985c75c91f;
 
+    bytes4 constant private AGENT_INIT_SELECTOR = 0x8129fc1c;
+    bytes4 constant private VAULT_INIT_SELECTOR = 0x8129fc1c;
+    bytes4 constant private FINANCE_INIT_SELECTOR = 0x1798de81;
+    bytes4 constant private VOTING_INIT_SELECTOR = 0xdf3d3305;
```

Comment:
Do we need this? Can't we do something like `abi.encodeWithSelector(Agent.initialize.selector)`?

Reply:
I tried to go that way, but the contract needs to be instantiated to be able to use `.selector`.